src/Raven.Client/Documents/Operations/Operation.cs

```diff
@@ -54,12 +54,25 @@ public Operation(RequestExecutor requestExecutor, Func<IDatabaseChanges> changes
             var observable = changes.ForOperationId(_id);
             _subscription = observable.Subscribe(this);
             await observable.EnsureSubscribedNow().ConfigureAwait(false);
-
+            changes.ConnectionStatusChanged += ConnectionStatusEvent;
             await FetchOperationStatus().ConfigureAwait(false);
         }
 
+        private async void ConnectionStatusEvent(object sender, EventArgs e)
```

Comment:
Let's avoid `async void` and friends.
And we need to call to `TrySetException` in this case, no?

Reply:
@aviviadi what do you think about this approach: https://github.com/ppekrol/ravendb/commit/5e32ddaf75c03955126c5db0ad0a8b4ca2d8ae56 ?

Reply:
@ayende - if we TrySetException, we might report failure on a successful operation. Hunged client (with hopefully timeout set) is more reasonable for operation with no knowledge if it went all the way or not (and the connection is probably "dead" anyhow at this point)

Reply:
@ppekrol - obviously that's better (fixes the above remark)

Reply:
Yes, let's go ahead and merge this